PageObjects/BasePage.cs

```diff
@@ -0,0 +1,36 @@
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using SwagLabsTask.Utilities;
+
+namespace SwagLabsTask.PageObjects
+{
+    public class BasePage
+    {
+        
+        protected readonly IWebDriver driver;
+        protected readonly WebDriverWait wait;
+        protected string Url {get;}
+
+        public BasePage(string browser, TimeSpan timeout, string Url)
+        {
+            this.driver = WebDriverManager.GetInstance(browser).Driver;
+            wait = new WebDriverWait(driver, timeout);
+            this.Url = Url;
+        }
+
+        public string GetCurrentUrl()
+        {
+            return driver.Url;
+        }
+
+        public string GetPageUrl()
+        {
+            return this.Url;
+        }
+
+        public void Close()
+        {
+            WebDriverManager.ResetInstance();
+        }
+    }
+}
```

Comment:
The class name "Base page" is usually used to initiate the elements in the derived classes as well, you can use it in the base class constructor and delete it from the derived page objects. This class looks like driver implementation, so should be in another folder like "Drivers" and should be renamed